pkgs/tools/admin/balena-cli/default.nix

```diff
@@ -1,94 +1,59 @@
 { lib
 , stdenv
-, fetchzip
+, buildNpmPackage
+, fetchFromGitHub
 , testers
+, balena-cli
+, nodePackages
+, python3
+, udev
+, darwin
+, patchelf
 }:
-let
-  inherit (stdenv.hostPlatform) system;
-  throwSystem = throw "Unsupported system: ${system}";
 
-  plat = {
-    x86_64-linux = "linux-x64";
-    x86_64-darwin = "macOS-x64";
-    # Balena only packages for x86 so we rely on Rosetta for Apple Silicon
-    aarch64-darwin = "macOS-x64";
-  }.${system} or throwSystem;
-
-  sha256 = {
-    x86_64-linux = "sha256-USljQ/cnbSabzsZWXlZ0eeZSqkTr3wVP0ktXqZ7Fw4U=";
-    x86_64-darwin = "sha256-NWzJPB+HzlsB6yTcEMwTg8pioonGWPOU96jyIpiZiTY=";
-    aarch64-darwin = "sha256-NWzJPB+HzlsB6yTcEMwTg8pioonGWPOU96jyIpiZiTY=";
-  }.${system} or throwSystem;
-
-  version = "16.7.6";
-  src = fetchzip {
-    url = "https://github.com/balena-io/balena-cli/releases/download/v${version}/balena-cli-v${version}-${plat}-standalone.zip";
-    inherit sha256;
-  };
-in
-stdenv.mkDerivation (finalAttrs: {
+buildNpmPackage rec {
   pname = "balena-cli";
-  inherit version src;
+  version = "17.0.0";
 
-  installPhase = ''
-    runHook preInstall
+  src = fetchFromGitHub {
+    owner = "balena-io";
+    repo = "balena-cli";
+    rev = "v${version}";
+    hash = "sha256-sNpxjSumiP+4fX6b3j+HEl/lr4pvudrhfTzr2TYastE=";
+  };
 
-    mkdir -p $out/bin
-    cp -r ./* $out/
-    ln -s $out/balena $out/bin/balena
+  npmDepsHash = "sha256-q2Yc6e5dEiP2Q4tFIeqj4mswM1/pX1pdGeoagyiupvs=";
 
-    runHook postInstall
+  postPatch = ''
+    ln -s npm-shrinkwrap.json package-lock.json
   '';
+  makeCacheWritable = true;
+
+  nativeBuildInputs = [
+    nodePackages.node-gyp
+    python3
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.cctools
+    # From some reason this is not included in stdenv of Darwin
+    patchelf
```

Comment:
```suggestion
```

Darwin binaries are in Mach-O format, not ELF, so `patchelf` should have no effect.

Reply:
> Darwin binaries are in Mach-O format, not ELF, so `patchelf` should have no effect...?

I wasn't sure, thanks for your help. Removed it.